### PR TITLE
[PIXELS-732] support one table scan

### LINKED ISSUE #118: Run worker coordinator service for pipline execution

### PR DESCRIPTION
**Pull request by huasiy**
Resolves #118

### CODE REVIEW

**Comment by huasiy** [comment]
image

**Comment by huasiy** [comment]
image

File: connector/pom.xml [hunk 1]
```diff
@@ -69,6 +69,11 @@
             <artifactId>pixels-invoker-lambda</artifactId>
             <optional>true</optional>
         </dependency>
+        <dependency>
+            <groupId>io.pixelsdb</groupId>
+            <artifactId>pixels-invoker-vhive</artifactId>
```
Comment (bianhq) [inline connector/pom.xml:74]: Remove this dependency. Multiple invoker dependencies will overwrite the invokers for a worker type.

To solve this problem, we need to modify the InvokerFactory in pixels-common and allow it to load multiple invokers for the same worker type into invokerMap.

File: connector/src/main/java/io/pixelsdb/pixels/trino/PixelsWorkerCoordinator.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package io.pixelsdb.pixels.trino;
```
Comment (bianhq) [inline connector/src/main/java/io/pixelsdb/pixels/trino/PixelsWorkerCoordinator.java:1]: Add a license header.

File: connector/src/main/java/io/pixelsdb/pixels/trino/PixelsWorkerCoordinator.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+import io.pixelsdb.pixels.common.utils.ConfigFactory;
+import io.pixelsdb.pixels.planner.coordinate.WorkerCoordinateServer;
+
+public class PixelsWorkerCoordinator {
```
Comment (bianhq) [inline connector/src/main/java/io/pixelsdb/pixels/trino/PixelsWorkerCoordinator.java:6]: Add a javadoc before class definition, which includes the usage/notice/design of this class and the `@author` and `@create` tags.
Add a new line before {.

File: connector/src/main/java/io/pixelsdb/pixels/trino/PixelsWorkerCoordinator.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    private static WorkerCoordinateServer instance;
+
+    private PixelsWorkerCoordinator() {}
+
+    public static void init()
+    {
+        if (instance == null) {
```
Comment (bianhq) [inline connector/src/main/java/io/pixelsdb/pixels/trino/PixelsWorkerCoordinator.java:13]: Add a new line before {.

File: connector/src/main/java/io/pixelsdb/pixels/trino/PixelsWorkerCoordinator.java [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+            int port = Integer.parseInt(ConfigFactory.Instance().getProperty("worker.coordinate.server.port"));
+            instance = new WorkerCoordinateServer(port);
+            Thread workerThread = new Thread(instance);
+            workerThread.start();
```
Comment (bianhq) [inline connector/src/main/java/io/pixelsdb/pixels/trino/PixelsWorkerCoordinator.java:17]: We should terminate the worker coordinator when the Trino process is shutting down.
I think we can try:
1. terminate the worker coordinator server in the shutdown method of PixelsConnector.
2. add a JVM shutdown hook here to terminate the worker coordinator server when the Trino process is shutting down.

**Review by bianhq — CHANGES_REQUESTED** [review]

File: connector/src/main/java/io/pixelsdb/pixels/trino/PixelsConnectorFactory.java [hunk 6]
```diff
@@ -22,17 +22,23 @@
...
+import org.slf4j.LoggerFactory;
 
 import java.util.Map;
 
 import static java.util.Objects.requireNonNull;
 
 public class PixelsConnectorFactory implements ConnectorFactory
 {
+    private static final Logger log = LoggerFactory.getLogger(PixelsConnectorFactory.class);
+    private WorkerCoordinateServer coordinateServer;
```
Comment (bianhq) [inline connector/src/main/java/io/pixelsdb/pixels/trino/PixelsConnectorFactory.java:41]: Is this coordinateServer needed?

**Review by bianhq — CHANGES_REQUESTED** [review]

**Review by bianhq — APPROVED** [review]
LGTM.

**Pull request merged**